cabal-install/src/Distribution/Client/ProjectConfig.hs

```diff
@@ -534,6 +531,15 @@ resolveBuildTimeSettings
         | isParallelBuild buildSettingNumJobs = False
         | otherwise = False
 
+-- | Determine the number of jobs (ParStrat) from the project config
+resolveNumJobsSetting :: Flag Bool -> Flag (Maybe Int) -> ParStratX Int
+resolveNumJobsSetting projectConfigUseSemaphore projectConfigNumJobs =
```

Comment:
Although the names of parameters are pretty descriptive, I'd prefer to have little haddocks for parameters that are so weakly typed (Boo, Int). Especially considering this is an exported function.